File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def goodNodes(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        return 1 + self.goodNodeWithMax(root.left, root.val) + self.goodNodeWithMax(root.right, root.val)
        
    def goodNodeWithMax(self, root, maxNum):
        if root is None:
            return 0
        
        count = 0
        if root.val >= maxNum:
            count = 1
            maxNum = root.val

        return count + self.goodNodeWithMax(root.left, maxNum) + self.goodNodeWithMax(root.right, maxNum)

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func goodNodesWithVal(root *TreeNode, val int) int {
    if root == nil {
        return 0
    }

    if root.Val >= val {
        return 1 + goodNodesWithVal(root.Left, root.Val) + goodNodesWithVal(root.Right, root.Val)
    }

    return goodNodesWithVal(root.Left, val) + goodNodesWithVal(root.Right, val)
}

func goodNodes(root *TreeNode) int {
    return goodNodesWithVal(root, root.Val)
}